### PR TITLE
test: post extension publish test

### LINKED ISSUE #2: Post extension publish CI to make sure published extension works
This repo should contain an e2e test for installing and using our VSCode extension. Should probably have 2 branches, `stable` and `insider` where we automatically install a version if it appears in the marketplace. Then some basic tests are run to double check that it works. On success or failure a message is posted to a #feed-... slack channel.

### PR DESCRIPTION
**Pull request by carmenberndt**
closes #2

### CODE REVIEW

**Comment by janpio** [comment]
Do I understand correctly that every 10 minutes we try to install the latest version, if we get a new version we run some tests and otherwise we stop and wait until the next run?

**Comment by carmenberndt** [comment]
> Do I understand correctly that every 10 minutes we try to install the latest version, if we get a new version we run some tests and otherwise we stop and wait until the next run?

Almost.
- Every 10 minutes we check if there is a new version on the marketplace.
- If so, we install the extension and run the tests
- Notification to Slack on test outcome

This is done for both the insider and stable version.

**Comment by janpio** [comment]
That what I was expecting, but did not understand how. Can you point me to "we check if there is a new version on the marketplace."?

File: .github/workflows/post_extension_publish_CI.yml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    if: endsWith(github.ref, '/master')
+    steps:
+      - uses: actions/checkout@v2
+      - name: Compile
+        run: npm run build
+      - name: Check new release of stable
```
Comment (carmenberndt) [inline .github/workflows/post_extension_publish_CI.yml:27]: @janpio This is the step checking the current version from the marketplace.

File: scripts/check-version.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+
+set -eu
+
+EXTENSION_TYPE=$1
+echo "EXTENSION_TYPE: $EXTENSION_TYPE"
+
+if [ "$EXTENSION_TYPE" = "insider" ]; then
+    TESTED_VERSION=$(cat scripts/tested_prisma_version_insider)
+    EXTENSION_VERSION=$(./node_modules/.bin/vsce show Prisma.prisma-insider --json | jq ".versions[0].version" | tr -d '"')
```
Comment (carmenberndt) [inline scripts/check-version.sh:10]: @janpio and here the version gets pulled via the `vsce show` command

File: scripts/check-version.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    EXTENSION_VERSION=$(./node_modules/.bin/vsce show Prisma.prisma --json | jq ".versions[0].version" | tr -d '"')
+fi
+echo "Last tested extension version: $TESTED_VERSION"
+
+echo "Currently published extension version: $EXTENSION_VERSION"
+
+echo "::set-output name=version::$EXTENSION_VERSION"
+
+if [ "$TESTED_VERSION" != "$EXTENSION_VERSION" ]; then
+    echo "::set-output name=new_version::true"
```
Comment (carmenberndt) [inline scripts/check-version.sh:22]: @janpio So only if the last stored version is not the same as the one from the marketplace, the rest of the steps will be executed.

**Pull request merged**